gobcore/standalone.py

```diff
@@ -0,0 +1,157 @@
+import argparse
+import json
+from gobcore.logging.logger import logger
+from pathlib import Path
+from typing import Dict, Any, Callable, Tuple
+
+from gobcore.utils import get_logger_name
+from gobcore.message_broker.offline_contents import offload_message, load_message
+from gobcore.message_broker.utils import to_json, from_json
+
+Message = Dict[str, Any]
+
+
+def parent_argument_parser() -> Tuple[argparse.ArgumentParser, argparse._SubParsersAction]:
+    """Setup parent argument parser, to which subparsers can be added.
+
+    Add any 'handler'-functions to the returned subparser, call parse_args() on
+    the parent parser when all handlers are added.
+
+    :return: The parent parser and the subparser, to add subcommands to.
+    """
+    parser = argparse.ArgumentParser(
+        description='Start standalone GOB Tasks',
+    )
+    parser.add_argument(
+        "--message-data",
+        required=False,
+        help="Message data used by the handler."
+    )
+    parser.add_argument(
+        "--message-result-path",
+        default="/airflow/xcom/return.json",
+        help="Path to store result message."
+    )
+    subparsers = parser.add_subparsers(
+        title="handlers",
+        help="Which handler to run.",
+        dest="handler",
+        required=True
+    )
+    return parser, subparsers
+
+
+def run_as_standalone(
+        args: argparse.Namespace, service_definition: dict[str, Any]
+) -> int:
+    """Runs application in standalone mode.
+
+    Finds the handler to run from the arguments given. For 'start commands' the
+    message is constructed from arguments, for example with a catalogue and
+    collection. 'Non-start commands' are instructed with a message received
+    from a handler called in a previous task.
+
+    :param args: Arguments as parsed by arg parse.
+    :param service_definition: A dict with keys which maps to handlers.
+    :return: the resulting message data from the handler.
+    """
+    message = _build_message(args)
+    print(f"Loading incoming message: {message}")
```

Comment:
A message is an implementation detail. When running standalone this is not an abstraction you necessarily know of - so this line doesn't make sense.. ;)

If I run **import --catalogue=nap --collection=peilmerken** for example. How does 'Loading incoming message' make sense to the user?